examples/validating-public-input/contracts/src/FibonacciValidator.sol

```diff
@@ -0,0 +1,79 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.12;
+
+contract FibonacciValidator {
+    address public alignedServiceManager;
+    bytes32 public fibonacciImageId;
+
+    bytes32 public fibonacciImageIdCommitment =
+        0xbfa561e384be753bd6fd75b15db31eb511cd114ec76d619a87c2342af0ee1ed7;
+
+    event FibonacciNumbers(uint32 fibN, uint32 fibNPlusOne);
+
+    constructor(address _alignedServiceManager) {
+        alignedServiceManager = _alignedServiceManager;
+    }
+
+    function verifyBatchInclusion(
+        bytes32 proofCommitment,
+        bytes32 pubInputCommitment,
+        bytes32 provingSystemAuxDataCommitment,
+        bytes20 proofGeneratorAddr,
+        bytes32 batchMerkleRoot,
+        bytes memory merkleProof,
+        uint256 verificationDataBatchIndex,
+        bytes memory journalBytes
+    ) public returns (bool) {
+        require(
+            fibonacciImageIdCommitment == provingSystemAuxDataCommitment,
+            "Image ID doesn't match"
+        );
+
+        require(
+            pubInputCommitment == keccak256(abi.encodePacked(journalBytes)),
+            "Fibonacci numbers don't match with public input"
+        );
+
+        (uint32 fibN, uint32 fibNPlusOne) = bytesToTwoUint32(journalBytes);
+
+        emit FibonacciNumbers(fibN, fibNPlusOne);
```

Comment:
maybe emit event after verification

Reply:
Solved in [eb6aa2e](https://github.com/yetanotherco/aligned_layer/pull/648/commits/eb6aa2eada3299e0b497f73848a57014b8fe3f74)